src/webgl/primitives.js

```diff
@@ -667,64 +766,21 @@ p5.RendererGL.prototype.triangle = function
     triGeom.computeNormals();
     this.createBuffers(gId, triGeom);
   }
-
   this.drawBuffers(gId);
   return this;
 };
 
-p5.RendererGL.prototype.ellipse = function
-(args){
-  var x = args[0];
-  var y = args[1];
-  var width = args[2];
-  var height = args[3];
-  //detailX and Y are optional 6th & 7th
-  //arguments
-  var detailX = args[4] || 24;
-  var detailY = args[5] || 16;
-  var gId = 'ellipse|'+args[0]+'|'+args[1]+'|'+args[2]+'|'+
-  args[3];
-  if(!this.geometryInHash(gId)){
-    var _ellipse = function(){
-      var u,v,p;
-      var centerX = x+width*0.5;
-      var centerY = y+height*0.5;
-      for (var i = 0; i <= this.detailY; i++){
-        v = i / this.detailY;
-        for (var j = 0; j <= this.detailX; j++){
-          u = j / this.detailX;
-          var theta = 2 * Math.PI * u;
-          if(v === 0){
-            p = new p5.Vector(centerX, centerY, 0);
-          }
-          else{
-            var _x = centerX + width*0.5 * Math.cos(theta);
-            var _y = centerY + height*0.5 * Math.sin(theta);
-            p = new p5.Vector(_x, _y, 0);
-          }
-          this.vertices.push(p);
-          this.uvs.push([u,v]);
-        }
-      }
-    };
-    var ellipseGeom = new p5.Geometry(detailX,detailY,_ellipse);
-    ellipseGeom
-      .computeFaces()
-      .computeNormals();
-    if(detailX <= 24 && detailY <= 16) {
-      ellipseGeom._makeTriangleEdges();
-      this._edgesToVertices(ellipseGeom);
-    } else {
-      console.log('Cannot stroke ellipse with more'+
-        ' than 24 detailX or 16 detailY');
-    }
-
-    this.createBuffers(gId, ellipseGeom);
-  }
-  this.drawBuffers(gId);
+p5.RendererGL.prototype.ellipse = function(args){
+  //Pass arguments X, Y, Width, Height, and detail to Arc
+  //Supply additional parameters directing Arc to draw a full ellipse
+  this.arc(args[0], args[1] ,args[2], args[3],
+           0, Math.PI * 2, 'ellipse', args[4]);
```

Comment:
great. can 'ellipse' be a constant? ok to add this in a future commit, but let's open an issue so we don't forget!